daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionCoder.scala

```diff
@@ -255,11 +255,16 @@ object TransactionCoder {
       _ = node.choiceObservers.foreach(builder.addObservers)
       _ <- node.choiceAuthorizers match {
         case Some(authorizers) =>
-          Either.cond(
-            node.version >= TransactionVersion.minChoiceAuthorizers,
-            authorizers.foreach(builder.addAuthorizers),
-            EncodeError(s"choice authorizers are not supported by ${node.version.protoValue}"),
-          )
+          if (node.version <= TransactionVersion.minChoiceAuthorizers)
+            Right(
```

Comment:
This should be a `Left` but this codepath isn't currently tested

```suggestion
            Left(
```